src/main/java/com/github/dariobalinzo/filter/BlacklistFilter.java

```diff
@@ -31,8 +31,8 @@ public BlacklistFilter(Set<String> fieldsToRemove) {
     private Object filterBlacklistItem(String key, Object value) {
         if (value instanceof Map || value instanceof List) {
             boolean shouldVisitNestedObj = fieldsToRemove.stream()
-                    .anyMatch(jsonPath -> jsonPath.startsWith(key));
-            return shouldVisitNestedObj ? value : null;
+                    .anyMatch(jsonPath -> key.startsWith(jsonPath));
```

Comment:
can you please provide a simple test where filtering a whole map/lists?
it should be easy, just take as reference the BlacklistFilterTest file and add some other new flles in the resources for comparison.

Thanks,
Dario